ipa-core/src/report.rs

```diff
@@ -110,6 +110,25 @@ impl From<&EventType> for u8 {
     }
 }
 
+impl From<&EventType> for Replicated<Boolean> {
+    fn from(value: &EventType) -> Self {
+        Replicated(
+            Boolean::from(*value == EventType::Trigger),
+            Boolean::from(*value == EventType::Trigger),
+        )
+    }
+}
+
+impl From<&Replicated<Boolean>> for EventType {
```

Comment:
Is this for tests? It doesn't seem safe to make this generally available.

Reply:
I agree. This is again caused that we don't separate OprfReport from InputRow. Report has event_type in the clear while InputRow requires it to be secret shared. Therefore, OprfReport has it secret shared. Right now the From function is only used to generate EncryptedOprfReports from OprfReports in a convenient way. It is called by the encrypt function from OprfReport. The encrypt function is only used in test and playbook_oprf_ipa when the plaintext_match_keys flag is set false. Encrypt is done by the device in real IPA, so for our server based implementation it is only used for testing purposes. However, I dont think I should add a test flag in front of playbook_oprf_ipa.